src/api/entities/Instruction/index.ts

```diff
@@ -173,6 +176,44 @@ export class Instruction extends Entity<UniqueIdentifiers, string> {
     return statusResult === InternalInstructionStatus.Pending;
   }
 
+  /**
+   * Retrieve current status of the Instruction. This can be subscribed to know if instruction fails
+   *
+   * @note can be subscribed to
+   */
+  public async onStatusChange(callback: SubCallback<InstructionStatus>): Promise<UnsubCallback> {
+    const {
+      context: {
+        polymeshApi: {
+          query: {
+            settlement: { instructionDetails },
+          },
+        },
+      },
+      id,
+      context,
+    } = this;
+
+    const assembleResult = (rawStatus: MeshInstructionStatus): InstructionStatus => {
+      const status = meshInstructionStatusToInstructionStatus(rawStatus);
+
+      if (status === InternalInstructionStatus.Unknown) {
+        throw new PolymeshError({
+          code: ErrorCode.DataUnavailable,
+          message: executedMessage,
+        });
+      }
```

Comment:
This isn't a specific way of knowing if Instruction was rejected/executed. One way to do so (when we receive status from chain as `Unknown`) can be to query middleware to check so, but maybe the middleware isn't all synced upto the block. 

In a hacky way maybe we add some delay in querying middleware but that's feel right. Subscription should update the status as soon as it changes otherwise the whole purpose gets destroyed. 

Let me know you thoughts on how you guys think we can handle the `Rejected/Executed` case @VictorVicente @polymath-eric

Reply:
As per the task DA-301, what it sounds is that `Failed` status was the one that was most concerning but since its a generic status change function, I think it should tell us about all

Reply:
Difference between being rejected and executed is very big to not be able to know it. We should discuss with Robert and team how would it be the best way to track it. 
One alternative could be using events, but it's not my preferred by default one. 
Maybe there is some alternative on the scheduler side. I'll kickoff a discussion with Robert to find the best solution here.

Reply:
I thought we able to subscribe to the events with polkadot and when we see a block with it we can resolve the promise / call the callback? I think the subscription can narrow to event type, but sure if we could subscribe to a specific instruction. Been a while since I worked on bridge watcher but it should have an example.

We shouldn't use the middleware for this since it can add latency and make things feel slow.